### PR TITLE
204 2023 12 06 previewing a question during creation

### PR DESCRIPTION
**Pull request by christophercameron-ixd**
## Preview

https://forms-prototypes-pr-208.herokuapp.com/

## What

- Added new [CTA design work from the Mural](https://app.mural.co/t/gaap0347/m/gaap0347/1698835640099/6343df85f7816e0f9d711bd8dc144cd004ea5bce?wid=0-1699284569672)
  - this includes simplifying the save call to actions to be a single action
  - added the new next steps to the success banner (Add/Edit next question) 
  - updated back links to show the page the user should be taken back to, reducing the likelihood of them just seeing “Back” without context
- Added new “Save and preview” button to “Check your question” page
- Added new “Preview question” page
  - includes a <div> showing what the question will look like to the end user minus any back/continue interactions as per the [new preview designs on Mural](https://app.mural.co/t/gaap0347/m/gaap0347/1684744172504/f76417fddf1f3af0411a7769286931d0298176de?wid=0-1691573668953)

## Why

Designs implemented to align with where the development environment is with regards to the new CTA and back link work. This will allow us to test this when we test the new preview question page as part of the add/edit a question journey with users in the next round of research [TBC].

## Still needs done

Once we have the content for each answer type for the preview question page, we will need to add this in. It will need to be implemented before testing.

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -1414,3 +909,7 @@ router.get('/prototype-admin/show-data', (req, res, next) => {
   next()
 })
 
+/* Use the routes file in pages for adding and editing questions */
+router.use('/pages', require('./views/form-designer/pages/\_routes'))
+
+module.exports = router
```
Comment (lfdebrux) [inline app/routes.js:915]: This line shouldn't be necessary any more 🤔 

```suggestion
```

File: app/views/form-designer/pages/_routes.js [hunk 2]
```diff
@@ -0,0 +1,823 @@
...
+    'previousPageLink': previousPageLink,
+    'previousPageText': previousPageText
+  })
+
+  console.log(array)
+
+  return array
+}
+
+module.exports = router
```
Comment (lfdebrux) [inline app/views/form-designer/pages/_routes.js:823]:
```suggestion
```

File: app/routes.js [hunk 3]
```diff
@@ -1414,3 +909,7 @@ router.get('/prototype-admin/show-data', (req, res, next) => {
   next()
 })
 
+/* Use the routes file in pages for adding and editing questions */
+router.use('/pages', require('./views/form-designer/pages/\_routes'))
```
Comment (lfdebrux) [inline app/routes.js:913]: It should be sufficient to just require the extra routes file

```suggestion
require('./views/form-designer/pages/_routes')
```

File: app/views/form-designer/pages/_routes.js [hunk 4]
```diff
@@ -0,0 +1,823 @@
+const govukPrototypeKit = require('govuk-prototype-kit')
+const router = govukPrototypeKit.requests.setupRouter()
```
Comment (lfdebrux) [inline app/views/form-designer/pages/_routes.js:2]:
```suggestion
const router = govukPrototypeKit.requests.setupRouter('/pages')
```

File: app/routes.js (hunk 1 above)
Comment (christophercameron-ixd) [reply app/routes.js:915]: Thanks for taking a look @lfdebrux... This is the only line I seem to be able to remove without it falling over

File: app/views/form-designer/pages/_routes.js (hunk 4 above)
Comment (christophercameron-ixd) [reply app/views/form-designer/pages/_routes.js:2]: Thanks for taking a look @lfdebrux. I’ve tried altering the route files, and no matter what way I do it if I don’t include all the lines as they are it doesn’t seem to work.

Comment (lfdebrux) [reply app/views/form-designer/pages/_routes.js:2]: Okay, ignore my suggestions then. I'm not sure why it's not working, but I won't mess with what you've got!

**Review by lfdebrux — APPROVED** [review]
Works for me, looks cool too!

**Comment by christophercameron-ixd** [comment]
> Works for me, looks cool too!

Amazing, thank you!

**Comment by hazalarpalikli** [comment]
I should have wrote the comment here, i had a look at this and it looks good to me @christophercameron-ixd

A couple of things I am not sure about though; 
- When click "add guidance" it seems to be show up as a title in the preview option. See image below. Is this how it supposed to be? 
- the link for the prototype doesn't seem to work for me, I assume the correct link needs to be this? https://forms-prototypes.herokuapp.com/ 
Screenshot 2023-12-18 at 15 36 53

**Comment by christophercameron-ixd** [comment]
@hazalarpalikli I’m not sure what you mean that screenshot looks correct to me IF you added the “Guidance on the title” as the title and the “xoxoxo” as the Markdown? 

And yeah once the PR has been merged the pr heroku URL will be removed from heroku

**Comment by hazalarpalikli** [comment]
@christophercameron-ixd ignore my last comment please, i am getting this mixed up with the 'preview guidance'.

**Pull request merged**